File: src/ru/gurps/generator/desktop/models/characters/CharactersSpecialization.java
package ru.gurps.generator.desktop.models.characters;

import ru.gurps.generator.desktop.config.Model;

public class CharactersSpecialization extends Model {
    public Integer id;
    public Integer characterId;
    public Integer specializationId;
    public Integer cost;
    public Integer level;

    public CharactersSpecialization() {
    }

    public CharactersSpecialization(Integer characterId, Integer specializationId, Integer cost, Integer level) {
        this.characterId = characterId;
        this.specializationId = specializationId;
        this.cost = cost;
        this.level = level;
    }
}

File: src/main/kotlin/ru/gurps/generator/desktop/models/characters/CharactersSpecialization.kt
package ru.gurps.generator.desktop.models.characters

import ru.gurps.generator.desktop.config.Model
import java.sql.ResultSet

class CharactersSpecialization : Model {
    override var id: Int = -1
    var characterId: Int = -1
    var itemId: Int = -1
    var skillId: Int = -1
    var cost: Int = 0
    var level: Int = 0

    constructor()
    constructor(characterId: Int, itemId: Int, skillId: Int, cost: Int, level: Int) {
        this.characterId = characterId
        this.itemId = itemId
        this.skillId = skillId
        this.cost = cost
        this.level = level
    }

    constructor(result: ResultSet) {
        id = result.getInt("id")
        characterId = result.getInt("characterId")
        itemId = result.getInt("itemId")
        skillId = result.getInt("skillId")
        cost = result.getInt("cost")
        level = result.getInt("level")
    }
}
